sky/data/storage.py

```diff
@@ -545,21 +539,25 @@ def _validate_local_source(local_source):
                         f'Supported paths: local, s3://, gs://. Got: {source}')
         return source, is_local_source
 
-    def _validate_storage_spec(self) -> None:
+    def _validate_storage_spec(
+        self,
+        name: Optional[str],
+        source: Union[Path, List[Path], None],
+    ) -> Tuple[str, Union[Path, List[Path], None]]:
```

Comment:
Any particular reason to have this method return values instead of updating `self.source` and `self.name` in place?

Reply:
It was a remnant for fixing the mypy error. Just reverted it. Thanks!